beacon/beacon.go

```diff
@@ -83,17 +86,38 @@ func (api *Server) Query(w http.ResponseWriter, r *http.Request) {
 }
 
 func parseInput(r *http.Request) (*variants.Query, error) {
-	var query variants.Query
-	query.RefName = r.FormValue("chromosome")
-	query.Allele = r.FormValue("allele")
 
-	coord, err := getFormValueInt(r, "coordinate")
-	if err != nil {
-		return nil, fmt.Errorf("parsing coordinate: %v", err)
+	switch r.Method {
+	case "GET":
+		var query variants.Query
+		query.RefName = r.FormValue("chromosome")
+		query.Allele = r.FormValue("allele")
+
+		coord, err := getFormValueInt(r, "coordinate")
+		if err != nil {
+			return nil, fmt.Errorf("parsing coordinate: %v", err)
+		}
+		query.Coord = coord
+
+		return &query, nil
+	case "POST":
+		var params struct {
+			RefName string `json:"chromosome"`
+			Allele  string `json:"allele"`
+			Coord   *int64 `json:"coordinate"`
+		}
+		body, _ := ioutil.ReadAll(r.Body)
+		if err := json.Unmarshal(body, &params); err != nil {
+			return nil, fmt.Errorf("decoding request body: %v", err)
+		}
+		return &variants.Query{
```

Comment:
Optional, but if you just add the json tags to the Query type we can skip the 'params' structure altogether...

Reply:
I'll leave it like this for now as in future MR the api objects are different from the variants objects